src/components/MapDetail.js

```diff
@@ -0,0 +1,69 @@
+import React, { useEffect, useState } from 'react';
+import io from 'socket.io-client';
+import { MapContainer, TileLayer, Marker, Popup } from 'react-leaflet';
+import 'leaflet/dist/leaflet.css'; // Import Leaflet's CSS
+
+import icon from 'leaflet/dist/images/marker-icon.png';
+import iconShadow from 'leaflet/dist/images/marker-shadow.png';
+import L from 'leaflet';
+
+let DefaultIcon = L.icon({
+    iconUrl: icon,
+    shadowUrl: iconShadow
+});
+
+L.Marker.prototype.options.icon = DefaultIcon;
+
+const MapDetail = () => {
+  const [trainData, setTrainData] = useState({});
+
+  useEffect(() => {
+    const socket = io('http://localhost:1337'); // Connect to your Node.js server on port 1337
```

Comment:
Skriva om kommentaren? Låter lite 'personligt', kanske "Connecting to the Node.js server"? Eller ta bort det helt?